File: python/day3/puzzle-1.py

def checkCollision(x, line):
  point = line[x-1]
  if point == "#":
    return 1
  else:
    return 0

import numpy as np
inputFile = open("input.txt")
input = np.array([])
for line in inputFile:
  input = np.append(input,line)
inputFile.close()
idxRoot = 0
slopeX = 3
slopeY = 1
curX = 1
curY = 1
txt = "There are {} collisions on this route."
collisions = 0
while idxRoot < len(input):
  line = input[idxRoot]
  if (curY > 0):
    collisions += checkCollision(curX, line)
  curY += slopeY
  curX += slopeX
  idxRoot += 1
print(txt.format(collisions))

File: php/day3/puzzle-1.php
<?php

function checkColl($x, $line) {
    $spot = substr($line, $x-1, 1);
    if ($spot == "#") {
        return 1;
    }
    else {
        return 0; 
    }
}
$validCount = 0;
$filename = 'input.txt';
$fp = @fopen($filename, 'r');
$slopeX = 1;
$slopeY = 2;
$curX = 1;
$curY = 0;
$collisions = 0;
$handle = fopen("input.txt", "r");
if ($handle) {
    while (($line = fgets($handle)) !== false) {
        if ($curY > 0) {
            $collisions += checkColl($curX, $line);
         }
         $curY += $slopeY;
         $curX += $slopeX;
    }
    fclose($handle);
} 
echo "There are " . $collisions . " collisions on this slope.\n";